### PR TITLE
91980

### PR DESCRIPTION
**Pull request by saimekala07**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [91980](https://devops.dchbx.org/redmine/issues/91980) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | 3 |
| Peer Review Story Points | n/a |
| Ticket Author | @saipraveen18 |

### CODE REVIEW

File: app/models/hbx_enrollment.rb [hunk 1]
```diff
@@ -985,6 +990,25 @@ def update_renewal_coverage
...
   end
 
+  def update_reinstate_coverage
+    return unless is_shop? && benefit_sponsorship.present? && sponsored_benefit_package.benefit_application.present?
+    reinstated_app = benefit_sponsorship.benefit_applications.detect{|app| app.active? && app.reinstated_id == sponsored_benefit_package.benefit_application.id}
+    return unless reinstated_app.present?
+    parent_reinstated_app = reinstated_app.parent_reinstate_application
+    return unless parent_reinstated_app.benefit_packages.map(&:id).include?(sponsored_benefit_package_id)
+    reinstated_package = reinstated_app.benefit_packages.where(title: sponsored_benefit_package.title).first
+    return unless reinstated_package.present? || self.terminated_on != parent_reinstated_app.end_on
```
Comment (raghuramg) [inline app/models/hbx_enrollment.rb:1000]: self.terminated_on != parent_reinstated_app.end_on seems like not possible. Can you provide use case where we need this check.

**Review by raghuramg — CHANGES_REQUESTED** [review]
Please verify the comments. thanks

File: app/models/hbx_enrollment.rb (hunk 1 above)
Comment (saimekala07) [reply app/models/hbx_enrollment.rb:1001]: In method hbx_enrollment#update_existing_shop_coverage, when shopping under benefit application terminated/termination pending, the shopped enrollment(self here) will be terminated with benefit application termination date.

Here is the logic:

```
benefit_application = sponsored_benefit_package.benefit_application
return unless benefit_application.present? && benefit_application.terminated_on.present?
term_or_cancel_enrollment(self, benefit_application.terminated_on, benefit_application.termination_reason)
```

hbx_enrollment#update_reinstate_coverage,

Above method used In the scenario where any changes made parent reinstated enrollment, reinstated enrollment will be canceled and new reinstated enrollment created based on the new enrollment purchased.

**Review by raghuramg — APPROVED** [review]
Please promote this for testing

**Pull request merged**